### ISSUE TITLE
Infinite loop causes build failure

### ISSUE THREAD

**Issue by gborland** [evernote/android-state#9]
I am getting an infinite loop causing a StackOverflowError in Gradle when using android-state with Kotlin.

I have these two lines of code:
```kotlin
@State  var budgetMonth: YearMonth? = null
@State  var selectedCategoryId: String? = null
```

When I compile, I get this error:
```
20:43:04.260 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] Caused by: java.lang.StackOverflowError
20:43:04.260 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter]   at com.sun.tools.javac.model.JavacElements.getTypeElement(JavacElements.java:107)
20:43:04.260 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter]   at com.sun.tools.javac.model.JavacElements.getTypeElement(JavacElements.java:61)
20:43:04.260 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter]   at com.evernote.android.state.StateProcessor.getInsertedType(StateProcessor.java:733)
20:43:04.260 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter]   at com.evernote.android.state.StateProcessor.getInsertedType(StateProcessor.java:756)
20:43:04.260 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter]   at com.evernote.android.state.StateProcessor.getInsertedType(StateProcessor.java:756)
20:43:04.260 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter]   at com.evernote.android.state.StateProcessor.getInsertedType(StateProcessor.java:729)
20:43:04.260 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter]   at com.evernote.android.state.StateProcessor.getInsertedType(StateProcessor.java:756)
...
```
The log continues like that for several hundred lines.

If I remove the `@State` annotation from those two lines, the build completes successfully.

Kotlin 1.1.0
android-state 1.04

**Comment by vRallev** [comment]
I cannot reproduce the issue. I used the latest version of the library and Kotlin 1.1 as well. Could you share `YearMonth` or better a sample that reproduces the issue?

**Comment by gborland** [comment]
It's [`org.threeten.bp.YearMonth` from the threetenbp project](https://github.com/ThreeTen/threetenbp/blob/master/src/main/java/org/threeten/bp/YearMonth.java), included via https://github.com/JakeWharton/ThreeTenABP.

```java
public final class YearMonth
        extends DefaultInterfaceTemporalAccessor
        implements Temporal, TemporalAdjuster, Comparable<YearMonth>, Serializable {
```

I'll try to isolate a sample.

**Comment by gborland** [comment]
It works if I restore the `@State` annotation on the `String` field, so it does seem to be `YearMonth` which is causing the problem.

**Comment by vRallev** [comment]
Yes, I can reproduce the issue now, thanks!

**Comment by vRallev** [comment]
It's fixed in version `1.0.5`. Thanks for reporting.
